mongoengine/queryset/base.py

```diff
@@ -915,7 +915,7 @@ def only(self, *fields):
         .. versionchanged:: 0.5 - Added subfield support
         """
         fields = {f: QueryFieldList.ONLY for f in fields}
-        self.only_fields = fields.keys()
+        self.only_fields = list(fields.keys())
```

Comment:
I'm very confused why `future.utils` offers `listitems` and `listvalues` and doesn't offer `listkeys` 😕 

In Python 2, doing the redundant `list(d.keys())` is almost 3x slower compared to just `d.keys()`:
```
In [14]: d = {'a': 1, 'b': 2, 'c': 3}

In [15]: d.keys()
Out[15]: ['a', 'c', 'b']

In [16]: %timeit d.keys()
The slowest run took 15.45 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 3: 123 ns per loop

In [17]: %timeit list(d.keys())
The slowest run took 5.85 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 326 ns per loop
```

That said, it's likely negligible given that we're working w/ I/O in this package...

Reply:
I've thought about it some more and, at least in this case, the performance difference is so negligible that it's not worth discussing. Will resolve this comment now :)